engine/src/desktop-stack.cpp

```diff
@@ -225,9 +225,18 @@ void MCStack::realize(void)
 
 MCRectangle MCStack::view_platform_getwindowrect() const
 {
-	MCRectangle t_rect;
-	MCPlatformGetWindowFrameRect(window, t_rect);
-	return t_rect;
+    MCRectangle t_frame_rect;
+	MCPlatformGetWindowFrameRect(window, t_frame_rect);
+    if (MClockscreen != 0)
+    {
+        MCRectangle t_content_rect, t_diff_rect;
+        MCscreen->platform_getwindowgeometry(window, t_content_rect);
+        t_diff_rect = MCU_subtract_rect(t_content_rect, t_frame_rect);
```

Comment:
I don't think MCU_subtract_rect is correc there - t_content_rect here is the window's current rect, and t_frame_rect is the window's current effective rect. The frame rect will contain the content rect, which gives you four margins - one for top, left, right and bottom. You then want to apply these margins to the rect of the stack - e.g. left of effective rect = rect.left + (t_content_rect.x - t_frame_rect.x).